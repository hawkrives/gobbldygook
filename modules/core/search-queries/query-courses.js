import filter from 'lodash/filter'
import { checkCourseAgainstQuery } from './check-course-against-query'

/**
 * Queries the database for courses.
 *
 * @param {Object} queryObj - the query
 * @param {Array<Course>} courses - the courses to query
 * @returns {Array<Course>} - the courses that matched the query
 */
export function queryCourses(queryObj, courses) {
	return filter(courses, c => checkCourseAgainstQuery(queryObj, c))
}
