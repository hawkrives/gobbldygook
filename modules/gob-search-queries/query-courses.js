// @flow

import filter from 'lodash/filter'
import {checkCourseAgainstQuery} from './check-course-against-query'
import type {Course} from '@gob/types'

/**
 * Queries the database for courses.
 *
 * @param {Object} queryObj - the query
 * @param {Array<Course>} courses - the courses to query
 * @returns {Array<Course>} - the courses that matched the query
 */
export function queryCourses(queryObj: Object, courses: Array<Course>) {
	return filter(courses, c => checkCourseAgainstQuery(queryObj, c))
}
