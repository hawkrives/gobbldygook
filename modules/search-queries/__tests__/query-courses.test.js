import { queryCourses } from '../query-courses'

describe('queryCourses', () => {
    it('queries a list of courses', () => {
        let query = {
            departments: ['AMCON'],
            year: [2013],
        }
        let courses = [{ departments: ['AMCON'], year: 2013 }]
        expect(queryCourses(query, courses)).toMatchSnapshot()
    })

    it('properly handles a list of five years', () => {
        let query = {
            year: ['$OR', 2010, 2011, 2012, 2013, 2014],
        }
        let courses = [
            { departments: ['AMCON'], year: 2013 },
            { departments: ['ASIAN'], year: 2012 },
        ]
        expect(queryCourses(query, courses)).toMatchSnapshot()
    })
})
