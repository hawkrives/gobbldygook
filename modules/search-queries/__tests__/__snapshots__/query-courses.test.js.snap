// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queryCourses properly handles a list of five years 1`] = `
Array [
  Object {
    "departments": Array [
      "AMCON",
    ],
    "year": 2013,
  },
  Object {
    "departments": Array [
      "ASIAN",
    ],
    "year": 2012,
  },
]
`;

exports[`queryCourses queries a list of courses 1`] = `
Array [
  Object {
    "departments": Array [
      "AMCON",
    ],
    "year": 2013,
  },
]
`;
